src/parser.rs

```diff
@@ -1177,7 +1179,7 @@ impl<'a> Parser<'a> {
             Token::Mul | Token::Div | Token::Mod | Token::StringConcat => Ok(40),
             Token::DoubleColon => Ok(50),
             Token::ExclamationMark => Ok(50),
-            Token::LBracket | Token::RBracket => Ok(10),
+            Token::LBracket | Token::RBracket => Ok(51),
```

Comment:
Can you explain the need for this change?

Reply:
For this query
```
SELECT nspname WHERE (nspname !~ '^pg_temp_' OR nspname = (pg_catalog.current_schemas(true))[0])
```
The precedence is 20, while parsing map access. `[` so it skipped  to parse the `[` which is wrong. Then testcase broke when I use this query `SELECT nspname WHERE (nspname !~ '^pg_temp_' OR nspname = (pg_catalog.current_schemas(true))[0])`

so I bumped up the precedence for `[`

Let me know If I did something wront 😓 

Here is the debug log when the precedence is 10.
```
DEBUG [sqlparser::parser] Parsing sql 'SELECT nspname WHERE (nspname !~ '^pg_temp_' OR nspname = (pg_catalog.current_schemas(true))[0])'...
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Identifier(Ident { value: "nspname", quote_style: None })
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() Word(Word { value: "WHERE", quote_style: None, keyword: WHERE })
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Identifier(Ident { value: "nspname", quote_style: None })
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() ExclamationMarkTilde
DEBUG [sqlparser::parser] next precedence: 20
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Value(SingleQuotedString("^pg_temp_"))
DEBUG [sqlparser::parser] predences 20
DEBUG [sqlparser::parser] get_next_precedence() Word(Word { value: "OR", quote_style: None, keyword: OR })
DEBUG [sqlparser::parser] next precedence: 5
DEBUG [sqlparser::parser] get_next_precedence() Word(Word { value: "OR", quote_style: None, keyword: OR })
DEBUG [sqlparser::parser] next precedence: 5
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Identifier(Ident { value: "nspname", quote_style: None })
DEBUG [sqlparser::parser] predences 5
DEBUG [sqlparser::parser] get_next_precedence() Eq
DEBUG [sqlparser::parser] next precedence: 20
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Value(Boolean(true))
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() RParen
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] prefix: Function(Function { name: ObjectName([Ident { value: "pg_catalog", quote_style: None }, Ident { value: "current_schemas", quote_style: None }]), args: [Unnamed(Expr(Value(Boolean(true))))], over: None, distinct: false })
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() RParen
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] prefix: Nested(Function(Function { name: ObjectName([Ident { value: "pg_catalog", quote_style: None }, Ident { value: "current_schemas", quote_style: None }]), args: [Unnamed(Expr(Value(Boolean(true))))], over: None, distinct: false }))
DEBUG [sqlparser::parser] predences 20
DEBUG [sqlparser::parser] get_next_precedence() LBracket
DEBUG [sqlparser::parser] next precedence: 10
DEBUG [sqlparser::parser] get_next_precedence() LBracket
DEBUG [sqlparser::parser] next precedence: 10
DEBUG [sqlparser::parser] Tok: true
DEBUG [sqlparser::parser] get_next_precedence() RParen
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] get_next_precedence() RParen
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] prefix: Nested(BinaryOp { left: BinaryOp { left: Identifier(Ident { value: "nspname", quote_style: None }), op: PGRegexNotMatch, right: Value(SingleQuotedString("^pg_temp_")) }, op: Or, right: BinaryOp { left: Identifier(Ident { value: "nspname", quote_style: None }), op: Eq, right: Nested(Function(Function { name: ObjectName([Ident { value: "pg_catalog", quote_style: None }, Ident { value: "current_schemas", quote_style: None }]), args: [Unnamed(Expr(Value(Boolean(true))))], over: None, distinct: false })) } })
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() EOF
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] Parsing sql 'SELECT nspname WHERE (nspname !~ '^pg_temp_' OR nspname = (pg_catalog.current_schemas(true))[0])'...
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Identifier(Ident { value: "nspname", quote_style: None })
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() Word(Word { value: "WHERE", quote_style: None, keyword: WHERE })
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Identifier(Ident { value: "nspname", quote_style: None })
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() ExclamationMarkTilde
DEBUG [sqlparser::parser] next precedence: 20
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Value(SingleQuotedString("^pg_temp_"))
DEBUG [sqlparser::parser] predences 20
DEBUG [sqlparser::parser] get_next_precedence() Word(Word { value: "OR", quote_style: None, keyword: OR })
DEBUG [sqlparser::parser] next precedence: 5
DEBUG [sqlparser::parser] get_next_precedence() Word(Word { value: "OR", quote_style: None, keyword: OR })
DEBUG [sqlparser::parser] next precedence: 5
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Identifier(Ident { value: "nspname", quote_style: None })
DEBUG [sqlparser::parser] predences 5
DEBUG [sqlparser::parser] get_next_precedence() Eq
DEBUG [sqlparser::parser] next precedence: 20
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] parsing expr
DEBUG [sqlparser::parser] prefix: Value(Boolean(true))
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() RParen
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] prefix: Function(Function { name: ObjectName([Ident { value: "pg_catalog", quote_style: None }, Ident { value: "current_schemas", quote_style: None }]), args: [Unnamed(Expr(Value(Boolean(true))))], over: None, distinct: false })
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() RParen
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] prefix: Nested(Function(Function { name: ObjectName([Ident { value: "pg_catalog", quote_style: None }, Ident { value: "current_schemas", quote_style: None }]), args: [Unnamed(Expr(Value(Boolean(true))))], over: None, distinct: false }))
DEBUG [sqlparser::parser] predences 20
DEBUG [sqlparser::parser] get_next_precedence() LBracket
DEBUG [sqlparser::parser] next precedence: 10
DEBUG [sqlparser::parser] get_next_precedence() LBracket
DEBUG [sqlparser::parser] next precedence: 10
DEBUG [sqlparser::parser] Tok: true
DEBUG [sqlparser::parser] get_next_precedence() RParen
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] get_next_precedence() RParen
DEBUG [sqlparser::parser] next precedence: 0
DEBUG [sqlparser::parser] prefix: Nested(BinaryOp { left: BinaryOp { left: Identifier(Ident { value: "nspname", quote_style: None }), op: PGRegexNotMatch, right: Value(SingleQuotedString("^pg_temp_")) }, op: Or, right: BinaryOp { left: Identifier(Ident { value: "nspname", quote_style: None }), op: Eq, right: Nested(Function(Function { name: ObjectName([Ident { value: "pg_catalog", quote_style: None }, Ident { value: "current_schemas", quote_style: None }]), args: [Unnamed(Expr(Value(Boolean(true))))], over: None, distinct: false })) } })
DEBUG [sqlparser::parser] predences 0
DEBUG [sqlparser::parser] get_next_precedence() EOF
DEBUG [sqlparser::parser] next precedence: 0
thread 'parse_map_access_expr' panicked at 'assertion failed: `(left == right)`
  left: `"SELECT nspname WHERE (nspname !~ '^pg_temp_' OR nspname = (pg_catalog.current_schemas(true))[0])"`,
 right: `"SELECT nspname WHERE (nspname !~ '^pg_temp_' OR nspname = (pg_catalog.current_schemas(true)))"`', src/test_utils.rs:95:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:517:5
   1: core::panicking::panic_fmt
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panicking.rs:101:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panicking.rs:140:5
   4: sqlparser::test_utils::TestedDialects::one_statement_parses_to
             at ./src/test_utils.rs:95:13
   5: sqlparser::test_utils::TestedDialects::verified_stmt
             at ./src/test_utils.rs:103:9
   6: sqlparser::test_utils::TestedDialects::verified_query
             at ./src/test_utils.rs:109:15
   7: sqlparser::test_utils::TestedDialects::verified_only_select
             at ./src/test_utils.rs:118:15
   8: sqlparser_postgres::parse_map_access_expr
             at ./tests/sqlparser_postgres.rs:750:18
   9: sqlparser_postgres::parse_map_access_expr::{{closure}}
             at ./tests/sqlparser_postgres.rs:676:1
  10: core::ops::function::FnOnce::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/ops/function.rs:227:5
  11: core::ops::function::FnOnce::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
test parse_map_access_expr ... FAILED
```